cpp/oneapi/dal/backend/communicator.hpp

```diff
@@ -0,0 +1,99 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/backend/common.hpp"
+#include "oneapi/dal/detail/communicator.hpp"
+
+namespace oneapi::dal::backend {
+
+/// Implementation of SPMD communicator for one-rank system
+class fake_spmd_communicator : public dal::detail::spmd_communicator {
```

Comment:
The main question is about naming. Can it be called smth like `mock_spmd_communicator`?

Reply:
This is not a mock

Reply:
1) Mock is common word for such things. Like Mock tests 
2) Maybe It's just my concerns but due to the common sense the word fake has more negative meaning

Reply:
Well, the words "mock", "fake" and also "stub" are very common in testing you may find them in any SW testing book.